agent/src/main/java/org/openremote/agent/protocol/http/HTTPProtocol.java

```diff
@@ -196,16 +205,16 @@ protected Invocation.Builder getRequestBuilder(String value) {
                 requestBuilder.headers(headers);
             }
 
-            if (dynamicQueryParameters) {
-                requestBuilder.property(QueryParameterInjectorFilter.DYNAMIC_VALUE_PROPERTY, value);
-            }
-
             return requestBuilder;
         }
 
         protected Invocation buildInvocation(Invocation.Builder requestBuilder, String value) {
             Invocation invocation;
 
+            // Dynamic time Injection
+            value = ProtocolUtil.doDynamicTimeReplace(value, this.currentMillisSupplier);
+            requestBuilder.property(DynamicTimeInjectionFilter.CURRENT_MILLIS_SUPPLIER_PROPERTY, this.currentMillisSupplier);
```

Comment:
Add a `dynamicTime` flag (like `dynamicQueryParameters`) and do a super quick string.contains(`{$time`) on the headers and query params and then only set `DynamicTimeInjectionFilter.CURRENT_MILLIS_SUPPLIER_PROPERTY`, this will avoid the filter doing lots of regex work for no reason (as most uses of the protocol won't be using dynamic time).

Reply:
Fixed, though I was kind of saddened by how the Jakarta API doesn't quite work well with requestContexts. I was expecting something that is a bit more developer-friendly. I would be happy to write some convoluted Predicate to make stuff a bit easier to work with for both headers and query parameters, but I assume that there are other, more pertinent matters to deal with 😅